File: cpp/0034_FindFirstAndLastPositionOfElementInSortedArray.cpp



class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int left=getFirst(nums,target);
        int right=getLast(nums,target);
        return right>=left? vector<int>{left, right} : vector<int>{-1, -1};
        
    }
    int getFirst(vector<int>& nums, int target)
    {
        int left=0,right=nums.size()-1;
        
        while(left<=right)
        {
            int mid=left+(right-left)/2;
            if(nums[mid]<target)
                left=mid+1;
            else
                right=mid-1;
        }
        return left;
    }
    int getLast(vector<int>& nums, int target)
    {
        int left=0,right=nums.size()-1;
        
        while(left<=right)
        {
            int mid=left+(right-left)/2;
            if(nums[mid]<=target)
                left=mid+1;
            else
                right=mid-1;
        }
        return right;
    }
};

File: python/0034_FindFirstAndLastPositionOfElementInSortedArray.py
class Solution(object):
    def searchRange(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        left=self.findFist(nums,target)
        right=self.findLast(nums,target)
        return [left,right] if left<=right else [-1,-1]
        
    def findFist(self, nums, target):
        left,right=0,len(nums)-1
        while left<=right:
            mid=left+(right-left)/2
            if nums[mid]<target:
                left=mid+1
            else:
                right=mid-1
        return left


    def findLast(self, nums, target):
        left,right=0,len(nums)-1
        while left<=right:
            mid=left+(right-left)/2
            if nums[mid]<=target:
                left=mid+1
            else:
                right=mid-1
        return right
